sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/OneDeployTests.java

```diff
@@ -49,4 +65,78 @@ public void canDeployZip() throws Exception {
         String response = curl("https://" + webAppName1 + ".azurewebsites.net/" + "helloworld/").getValue();
         Assertions.assertTrue(response.contains("Hello"));
     }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    public void canPushDeployJar() throws Exception {
+        String webAppName1 = generateRandomResourceName("webapp", 10);
+
+        WebApp webApp1 =
+            appServiceManager
+                .webApps()
+                .define(webAppName1)
+                .withRegion(Region.US_WEST3)
+                .withNewResourceGroup(rgName)
+                .withNewLinuxPlan(PricingTier.STANDARD_S1)
+                .withBuiltInImage(RuntimeStack.JAVA_11_JAVA11)
+                .withHttpsOnly(true)
+                .create();
+
+        // deploy
+        File jarFile = new File("gateway.jar");
+        if (!jarFile.exists()) {
+            HttpURLConnection connection = (HttpURLConnection) new URL(GATEWAY_JAR_URL).openConnection();
+            connection.connect();
+            try (InputStream inputStream = connection.getInputStream();
+                 OutputStream outputStream = new FileOutputStream(jarFile)) {
+                IOUtils.copy(inputStream, outputStream);
+            }
+            connection.disconnect();
+        }
+        KuduDeploymentResult deployResult =
+            webApp1.pushDeploy(DeployType.JAR, jarFile, new DeployOptions().withTrackDeployment(true));
+
+        Assertions.assertNotNull(deployResult.deploymentId());
+
+        // poll deployment status
+        String deploymentStatusUrl = AzureEnvironment.AZURE.getResourceManagerEndpoint()
+            + "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/deploymentStatus/{deploymentId}?api-version=2021-03-01";
```

Comment:
I noticed they have a swagger for getting deployment by deploymentId, which includes deployment status as a part of the response: 
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/web/resource-manager/Microsoft.Web/stable/2021-02-01/WebApps.json#L2880-L2927

Will it help simplify?

Reply:
It appears not a same resource.

This one looks like below
```
{ 
  "id": "freeberg-java-deploy-test", 
  "name": "Microsoft.Web/sites", 
  "type": "Central US EUAP", 
  "properties": { 
    "id": "6e659b28-d0b0-4fd1-b216-e5655f238777", 
    "buildStatus": "RuntimeStarting", 
    "numberOfInstancesInProgress": 0, 
    "numberOfInstancesSuccessful": 6, 
    "numberOfInstancesFailed": 2, 
    "failedInstancesLogs": null 
  } 
} 
```